src/Microsoft.OData.Edm/Validation/ExpressionTypeChecker.cs

```diff
@@ -123,6 +123,8 @@ public static bool TryCast(this IEdmExpression expression, IEdmTypeReference typ
                     return TestTypeReferenceMatch(((IEdmCastExpression)expression).Type, type, expression.Location(), matchExactly, out discoveredErrors);
                 case EdmExpressionKind.LabeledExpressionReference:
                     return TryCast(((IEdmLabeledExpressionReferenceExpression)expression).ReferencedLabeledExpression, type, out discoveredErrors);
+                case EdmExpressionKind.EnumMember:
+                    return TryCastEnumConstantAsType((IEdmEnumMemberExpression)expression, type, matchExactly, out discoveredErrors);

```

Comment:
>expression [](start = 79, length = 10)

Is there any chance that type is an enum, but expression is not an IEdmEnumMemberExpression? If so, TryCastEnumConstantAsType will throw a null ref exception.